TestProject/output/BaseOutput.php

```diff
@@ -0,0 +1,52 @@
+<?php
+namespace output;
+
+use options\Getopt;
+use cellularAutomat\Board;
+
+/**
+ * BaseOutput class which provides the functions for pluggable-output.
+ *
```

Comment:
Please add a description what a developer must do to implement their own output. (You should for example explain which methods must be implemented, etc.)

Reply:
The documentation you added is very specific for different the different output types you already implemented. Please try to think more generic and from the perspective of the base class. So explain which methods exist that can be implemented and explain where in the lifecycle of the program execution they are called so that the developer knows when to use which method to implement certain things.
If you have questions about this I'm glad to clarify that to you, just ask!